### PR TITLE
jenkins_plugin: HTTP Error 405: Method Not Allowed on disable/enable plugin #2510

### LINKED ISSUE #2510: jenkins_plugin: HTTP Error 405: Method Not Allowed on disable/enable plugin
### Summary

When I use `state: disabled` for jenkins_plugin module for any plugin ansible fails with 405 response code, because jenkins_plugin.py sends GET request to jenkins but jenkins expects POST method for pluginManager/plugin/*  api requests.

I think this issue can easily be fixed with adding method=POST perameter to the _get_url_data here:
https://github.com/ansible-collections/community.general/blob/main/plugins/modules/web_infrastructure/jenkins_plugin.py#L696 

### Issue Type

Bug Report

### Component Name

jenkins_plugin

### Ansible Version

```console (paste below)
$ ansible --version
ansible 2.9.18
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug 13 2020, 02:51:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

RHEL 7

### Steps to Reproduce

```yaml (paste below)
- name: Install Plugins
  jenkins_plugin:
    name: xvfb
    url: "{{ jenkins_instance_url }}"
    state: disabled
```

### Expected Results

correctly disabled or enabled plugin

### Actual Results

```console (paste below)
TASK [jenkins : Disable or Enable Plugins] *************************************
failed: [test-jenkins-master-001] (item={'key': 'xvfb', 'value': 'false'}) => changed=false
  ansible_facts:
    discovered_interpreter_python: /usr/bin/python
  ansible_loop_var: item
  details: 'HTTP Error 405: Method Not Allowed'
  item:
    key: xvfb
    value: 'false'
  msg: Plugin not found.
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by brainsam**
Jenkins makeEnable/makeDisable api requests requires to use POST method

##### SUMMARY

Fixes #2510 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

jenkins_plugin

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).

File: changelogs/fragments/2510-jenkins_plugin_use_post_method.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - jenkins_plugin - Use POST method for enable/disable request to jenkins api (https://github.com/ansible-collections/community.general/issues/2510)
```
Comment (aminvakil) [inline changelogs/fragments/2510-jenkins_plugin_use_post_method.yml:2]:
```suggestion
  - jenkins_plugin - Use POST method for sending request to jenkins API when ``state`` option is ``enabled`` or ``disabled`` (https://github.com/ansible-collections/community.general/issues/2510).
```

File: plugins/modules/web_infrastructure/jenkins_plugin.py [hunk 2]
```diff
@@ -696,7 +696,8 @@ def _pm_query(self, action, msg):
         self._get_url_data(
             url,
             msg_status="Plugin not found. %s" % url,
-            msg_exception="%s has failed." % msg)
+            msg_exception="%s has failed." % msg,
+            method="POST")
```
Comment (aminvakil) [inline plugins/modules/web_infrastructure/jenkins_plugin.py:700]: I'm not sure, but this function is called for `uninstall` and `pinning` as well, how does this change affect them?

**Review by aminvakil — COMMENTED** [review]
Thanks for your contribution!

File: plugins/modules/web_infrastructure/jenkins_plugin.py (hunk 2 above)
Comment (brainsam) [reply plugins/modules/web_infrastructure/jenkins_plugin.py:700]: Hi @aminvakil, thx for your review

Jenkins requires POST requests for enable, disable and uininstall plugins
Pinning plugins feature is not supported in Jenkins 2 https://wiki.jenkins.io/display/JENKINS/Pinned+Plugins

Comment (aminvakil) [reply plugins/modules/web_infrastructure/jenkins_plugin.py:700]: > Jenkins requires POST requests for enable, disable and uininstall plugins

Please add `absent` to changelog fragment as well then.
> Pinning plugins feature is not supported in Jenkins 2 https://wiki.jenkins.io/display/JENKINS/Pinned+Plugins

I couldn't find anything in documentation that Jenkins < 2 is not supported, also as `pinned` is still a choice in this module for `state` option, it's better to fix that too IMHO (if it needs fixing).

Comment (brainsam) [reply plugins/modules/web_infrastructure/jenkins_plugin.py:700]: > I couldn't find anything in documentation that Jenkins < 2 is not supported,

https://www.jenkins.io/blog/2016/07/07/jenkins-2/

Comment (aminvakil) [reply plugins/modules/web_infrastructure/jenkins_plugin.py:700]: Sorry if I wasn't clear, I meant module documentation.
Even if jenkins 1 is EOL and unsupported, it has not been stated in this module, either change this to support `uninstalled` and `pinned` states or this should be marked as a `breaking_change` and has to wait until 4.0.0.

File: changelogs/fragments/2510-jenkins_plugin_use_post_method.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - jenkins_plugin - Use POST method for sending request to jenkins API when ``state`` option is ``enabled`` or ``disabled`` (https://github.com/ansible-collections/community.general/issues/2510).
```
Comment (felixfontein) [inline changelogs/fragments/2510-jenkins_plugin_use_post_method.yml:2]:
```suggestion
  - jenkins_plugin - use POST method for sending request to jenkins API when ``state`` option is one of ``enabled``, ``disabled``, ``pinned``, ``unpinned``, or ``absent`` (https://github.com/ansible-collections/community.general/issues/2510).
```

File: plugins/modules/web_infrastructure/jenkins_plugin.py (hunk 2 above)
Comment (felixfontein) [reply plugins/modules/web_infrastructure/jenkins_plugin.py:700]: I've added a suggestion for this below.

**Comment by felixfontein** [comment]
(Also please amend your last commit to remove the `@` from the commit message.)

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@brainsam thanks for fixing this!
@aminvakil thanks for reviewing!

**Pull request merged**